src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveTypeBuilderTests.cs

```diff
@@ -515,9 +515,9 @@ public void SaveMultipleGenericTypeParametersToEnsureSortingWorks()
                 saveMethod.Invoke(assemblyBuilder, new object[] { file.Path });
 
                 Module m = AssemblySaveTools.LoadAssemblyFromPath(file.Path).Modules.First();
-                Type[] type1Params = m.GetTypes()[2].GetGenericArguments();
+                Type[] type1Params = m.GetTypes()[0].GetGenericArguments();
```

Comment:
Why were the items reversed before?

Reply:
Before the type tokens created on `TypeBuilder.CreateType()` and before saving the types are ordered with the token order, the types here created in inverse order, you can see that above on row 512-514. 

Now with a global type that has global methods/fields this ordering/token assignment not working anymore, now all types and its members tokens created before save (therefore in defined order no matter when `TypeBuilder.CreateType()` called)